From: Mike Hommey <mh+mozilla@glandium.org>
Date: Fri, 4 Nov 2016 17:06:51 +0900
Subject: Bug 1115944 - Work around race condition between building NSPR and
 NSS. r=mshal

---
 config/recurse.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config/recurse.mk b/config/recurse.mk
index 2877c71945a8..f83230516ca8 100644
--- a/config/recurse.mk
+++ b/config/recurse.mk
@@ -172,6 +172,9 @@ ifeq ($(MOZ_REPLACE_MALLOC_LINKAGE),dummy library)
 mozglue/build/target memory/replace/logalloc/replay/target: memory/replace/dummy/target
 endif
 endif
+ifeq (,$(MOZ_SYSTEM_NSS)$(MOZ_FOLD_LIBS))
+config/external/nss/target: config/external/nspr/pr/target config/external/nspr/ds/target config/external/nspr/libc/target
+endif
 # Most things are built during compile (target/host), but some things happen during export
 # Those need to depend on config/export for system wrappers.
 $(addprefix build/unix/stdc++compat/,target host) build/clang-plugin/target: config/export
